Added tooltips to icons and settings.

ankur12-1610 opened the pull request:
Fixes: #775

This PR adds tooltips to icons and settings and all the changes are associated with `www/index.html` file.

![label-light](https://user-images.githubusercontent.com/76884959/145453610-6be734dc-fa3f-45ba-aca5-19e536222f75.png)
![label-dark](https://user-images.githubusercontent.com/76884959/145453751-52e716c9-288d-4f32-82ba-d5be5c1effb7.png)

ankur12-1610 commented:
I have added labels although I'm not sure that the content in the label is upto the mark or not :sweat_smile:.

Can anyone verify the changes?

Jaifroid commented:
I haven't forgotten this PR. I'll review it next.

ankur12-1610 commented:
> I haven't forgotten this PR. I'll review it next.

Sure, np :)

Jaifroid commented:
I've had a quick check on Chromium and Firefox, and the tooltips are showing fine on the Config page. I think we need to think about some descriptions that don't just repeat information already visible on the page. You have done this in some cases, but in others the information is basically the same. Perhaps in cases where it's obvious what a checkbox does, we don't need a tooltip?

I think we need a tooltip for the Random icon on the Home page, and in fact it was a comment from a user who didn't know what that button did that gave rise to this issue.

ankur12-1610 commented:
> I've had a quick check on Chromium and Firefox, and the tooltips are showing fine on the Config page. I think we need to think about some descriptions that don't just repeat information already visible on the page. You have done this in some cases, but in others the information is basically the same. Perhaps in cases where it's obvious what a checkbox does, we don't need a tooltip?
> 
> I think we need a tooltip for the Random icon on the Home page, and in fact it was a comment from a user who didn't know what that button did that gave rise to this issue.

Ah! okay got the point, but I'm not sure from which source should I take the information. Should I decide it by my own self or `Kiwix` has a trusted source from which it gathers the information?

Jaifroid commented:
The only documentation we have is what is in the "About" page of the app. But things which are obvious won't be documented there. I noticed you took some information from there for some options, and I agree with that approach.

I would say, please do the Random button, the Find/Search button, and if possible, the search field. I'll have a think about whether there is anything useful we could put in the tooltips that currently repeat info from the UI and get back to you.

ankur12-1610 commented:
> The only documentation we have is what is in the "About" page of the app. But things which are obvious won't be documented there. I noticed you took some information from there for some options, and I agree with that approach.
> 
> I would say, please do the Random button, the Find/Search button, and if possible, the search field. I'll have a think about whether there is anything useful we could put in the tooltips that currently repeat info from the UI and get back to you.

Okay got it, so for a time being I'll do it for the buttons, search field and other random features :+1:

Jaifroid commented:
@ankur12-1610 Here are my suggested edits (the first one in each pair is yours, the second one marked *** is what I suggest as a replacement):

```
1. <label title="Checking this option hides active content warning permanently." >
    *** <label title="A warning is shown if you load a ZIM that has active or dynamic content while you are in JQuery mode. It is not recommended to disable this warning." >
2.  <label title="Checking this option activates the animation transition between the app pages." >
    *** <label title="The animation is on by default and can be disabled by unchecking this option." >
3. <label title="This enables user to use home key to focus search bar.">
    *** <label title="This enables quick access to search, no matter where you are in an article.">
4. label title="This is used to select various themes like light, Dark (app only), Dark + invert (generic) and Dark + invert (Wikimedia).">
    *** <label title="This allows selection of themes either for the app only, or for the app and the loaded content.">
5. <label title="Cache is a high-speed data storage layer which stores a subset of data. This option enables caching assets which speeds up the display of articles.">
    *** <label title="The cache provides high-speed access to ZIM assets such as stylesheets and scripts, which will improve article load times.">
6. <label title="For low-memory devices caching assets is not possible and is not recommended. This option disables caching assets.">
    *** <label title="For low-memory devices, caching assets may cause app instability. This option allows you to disable the cache.">

```

ankur12-1610 commented:
> @ankur12-1610 Here are my suggested edits (the first one in each pair is yours, the second one marked *** is what I suggest as a replacement):
> 
> ```
> 1. <label title="Checking this option hides active content warning permanently." >
>     *** <label title="A warning is shown if you load a ZIM that has active or dynamic content while you are in JQuery mode. It is not recommended to disable this warning." >
> 2.  <label title="Checking this option activates the animation transition between the app pages." >
>     *** <label title="The animation is on by default and can be disabled by unchecking this option." >
> 3. <label title="This enables user to use home key to focus search bar.">
>     *** <label title="This enables quick access to search, no matter where you are in an article.">
> 4. label title="This is used to select various themes like light, Dark (app only), Dark + invert (generic) and Dark + invert (Wikimedia).">
>     *** <label title="This allows selection of themes either for the app only, or for the app and the loaded content.">
> 5. <label title="Cache is a high-speed data storage layer which stores a subset of data. This option enables caching assets which speeds up the display of articles.">
>     *** <label title="The cache provides high-speed access to ZIM assets such as stylesheets and scripts, which will improve article load times.">
> 6. <label title="For low-memory devices caching assets is not possible and is not recommended. This option disables caching assets.">
>     *** <label title="For low-memory devices, caching assets may cause app instability. This option allows you to disable the cache.">
> ```

@Jaifroid sorry for being inactive about a week, I was sick so couldn't work on it:sweat_smile:. I'll hop on it!

Thanks :)

Jaifroid commented:
Sorry to hear that, I hope you're feeling better!

ankur12-1610 commented:
> Sorry to hear that, I hope you're feeling better!

Yeah, now I'm fine. Thanks for asking :)

I've added tooltips to the Random button, Search button and Search field. Also, I've modified the content as per your instructions.

Kindly review.

Jaifroid commented on www/index.html:
```diff
@@ -79,13 +79,13 @@
                                 <span class="input-group">
-                                    <input type="search" id="prefix" placeholder="Search..." class="form-control" />
-                                    <button type="button" class="btn btn-outline-secondary" id="searchArticles"><i class="fas fa-search"></i></button>
+                                    <input type="search" title="Search field is the textfield which accepts user input to be searched for in a database." id="prefix" placeholder="Search..." class="form-control" />
```
Change to: "Search for article titles"

Jaifroid commented on www/index.html:
```diff
@@ -79,13 +79,13 @@
-                                    <input type="search" id="prefix" placeholder="Search..." class="form-control" />
-                                    <button type="button" class="btn btn-outline-secondary" id="searchArticles"><i class="fas fa-search"></i></button>
+                                    <input type="search" title="Search field is the textfield which accepts user input to be searched for in a database." id="prefix" placeholder="Search..." class="form-control" />
+                                    <button type="button" title="Search button's role is to search the query typed in the Seach field." class="btn btn-outline-secondary" id="searchArticles"><i class="fas fa-search"></i></button>
```
Change to: "Search"

Jaifroid commented on www/index.html:
```diff
@@ -435,29 +435,29 @@ <h3>Display settings</h3>
                                 </div>
                                 <div class="checkbox">
-                                    <label>
+                                    <label title="This enables quick access to search, no matter where you are in an article.">
```
Remove "This", so it reads "Enables quick access............"

Jaifroid commented on www/index.html:
```diff
@@ -435,29 +435,29 @@ <h3>Display settings</h3>
                                 </div>
                                 <div class="form-group">
-                                    <label>
+                                    <label title="This allows selection of themes either for the app only, or for the app and the loaded content.">
```
Remove "This" so it reads "Allows selection of..........................."

Jaifroid reviewed (changes requested):
@ankur12-1610 Thanks for the changes, just some small modifications, please.,

Jaifroid commented:
@ankur12-1610 Sorry about the changes to my suggested text. When testing it, starting some of them with "This" seemed redundant.

ankur12-1610 commented:
> This allows selection of themes either for the app only, or for the app and the loaded content.

Yeha, np :)

ankur12-1610 commented:
> @ankur12-1610 Sorry about the changes to my suggested text. When testing it, starting some of them with "This" seemed redundant.



> 
@Jaifroid Done :)

ankur12-1610 commented:
> @ankur12-1610 Sorry about the changes to my suggested text. When testing it, starting some of them with "This" seemed redundant.

@Jaifroid sorry for disturbing :sweat_smile:, can you review the PR?

Jaifroid reviewed (approved):
This all looks good to me know. Thanks!
@mossroy are you happy for this to be squashed and merged?

mossroy reviewed (approved):
OK for me

Jaifroid commented:
@ankur12-1610 Thank you for this PR. I've now merged it to master.
